### ISSUE TITLE
Square Tab Corners is broken

### ISSUE THREAD

**Issue by astyled** [BrowserWorks/waterfox#2940]
### What happened?

There is an option `Square Tab Corners` under `Tab Bar` section of `Look & Feel` and it works incorrectly.

When it's checked, bottom tab corners are still rounded after browser start. It's easier to explain with pictures:

![image](https://user-images.githubusercontent.com/42794343/209156025-e64a75c3-b8fc-4081-be2a-9169a57afd11.png)
(Squaring is **enabled** but browser is just started and I didn't touched that setting)

![image](https://user-images.githubusercontent.com/42794343/209156259-bf8e56fe-958e-49c1-a1fb-007357c15f89.png)
(Squaring is **disabled** but browser was not restarted yet)

![image](https://user-images.githubusercontent.com/42794343/209156421-32df8e31-1388-450e-b511-e2e2cbf4730c.png)
(Squaring is **enabled** again without browser restart)

After that, if I restart the browser the tabs will be again squared only partially.

This is actually looks good so it would be nice if this bug turned into feature.

```
Waterfox G5.1.1 (64-bit)
```

### Reproducible?

- [ ] I have tested this on Firefox.

### Version

Other

### What platform are you seeing the problem on?

Windows

**Comment by DavidSchillinger** [comment]
In case it helps, I see a similar issue with this setting except browser restart doesn't seem to affect it for me. (G5.1.2, Windows 11)

1. Settings -> Enable Square Tab Corners | Corners are now square even after a restart
3. Close browser settings | Corners remain square
4. Open browser settings (I do nothing else) | Bottom corners are now rounded
5. Settings -> Disable and re-enable Square Tab Corners | See 1.

Some sort of state persistence issue with the styles/settings menu?

**Comment by a0eoc** [comment]
Good catch. (For me too) the option will only apply correctly if the browser is restarted without the Settings tab and breaks again after opening Settings.

Honestly I see the best fix for this little issue to Square Tab Corners option be separated into options Square **Top** and Square **Bottom** Tab Corners. Not sure if that's possible or will bloat options too much but _square top_ + _rounded bottom_ looks best to me.

**Comment by MrAlex94** [comment]
Automated triage: This report was created before the Waterfox 6.7 update and has not been verified using the latest version. If you are still experiencing this issue with Waterfox 6.7, please comment with details of your version so that the issue can be reopened.
